apps/core/lib/core/event_repo.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Core.EventRepo do
+  @type operation ::
+          :user_created
+          | :user_updated
+          | :user_deleted
+          | :post_created
+
+@spec save_event(operation, {pos_integer(), String.t(), any()}) :: :ok | {:error, any()}
+  def save_event(:user_created, {user_id, user_email, time}),
+    do: write_to_mnesia(user_id, :user_created, %{user_email: user_email}, time)
+
+  def save_event(:user_updated, {user_id, attrs, time}),
+    do: write_to_mnesia(user_id, :user_updated, attrs, time)
+
+  def save_event(:user_deleted, {user_id, time}),
+    do: write_to_mnesia(user_id, :user_deleted, %{}, time)
+
+  def save_event(:post_created, {user_id, post_id, post_body, time}),
+    do:
+      write_to_mnesia(
+        user_id,
+        :post_created,
+        %{user_id: user_id, post_id: post_id, post_body: post_body},
+        time
+      )
+
+  def save_event(_, _), do: {:error, :invalid_operation}
+
+  defp write_to_mnesia(user_id, operation, attrs, time) do
+    case :mnesia.transaction(fn ->
+           :mnesia.write({Event, get_next_index(), user_id, operation, attrs, time})
+         end) do
+      {:atomic, :ok} -> :ok
+      {:aborted, reason} -> {:error, reason}
+    end
+  end
+
+  @spec get_event(pos_integer() | operation) :: {:ok, [list(keyword())]} | {:error, any()}
```

Comment:
Using `any` should be avoided whenever possible. Is there any list of possible errors from mnesia? Can we reuse it somehow here?

Reply:
I will check the specs, but some docs stated that 

> The function returns ok, or terminates if an error occurs. For example, the transaction terminates if no person table exists.

From meddling with mnesia in iex, this will propably be just an atom explaining what happened.

Reply:
> an atom explaining what happened.

so we could use a more strict typespec here ;)